### PR TITLE
[Packaging] Update Dockerfile base image from Alpine 3.14 to 3.15

### PR DESCRIPTION
**Pull request by danvy**
**Description**

Keep the pace with Alpine and some tools are only available on Alpine 3.15, ie : postgresql14-client

**Testing Guide**

"docker build -t azure-cli ." to check the build

**History Notes**

Dockerfile is now based on python Alpine 3.15

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Packaging

**Comment by jiasli** [comment]
Ops, the pipeline failed:

https://dev.azure.com/azure-sdk/public/_build/results?buildId=1319619&view=logs&j=a3141784-dfa9-5792-78cc-844daee2d774&t=b2b6de04-64e4-5812-4fff-06c02d19f4c2&l=35

```
Step 1/15 : ARG PYTHON_VERSION="3.9.6"
Step 2/15 : FROM python:${PYTHON_VERSION}-alpine3.15
manifest for python:3.9.6-alpine3.15 not found: manifest unknown: manifest unknown
```

We need to use a valid version from https://hub.docker.com/_/python.

I think we can use `3.10.2-alpine3.15` which will solve https://github.com/Azure/azure-cli/issues/19857#issuecomment-1016449222 by the way.

File: Dockerfile [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 ARG PYTHON_VERSION="3.9.6"
```
Comment (jiasli) [inline Dockerfile:6]:
```suggestion
ARG PYTHON_VERSION="3.10.2"
```

**Comment by danvy** [comment]
> Ops, the pipeline failed:
> 
> https://dev.azure.com/azure-sdk/public/_build/results?buildId=1319619&view=logs&j=a3141784-dfa9-5792-78cc-844daee2d774&t=b2b6de04-64e4-5812-4fff-06c02d19f4c2&l=35
> 
> ```
> Step 1/15 : ARG PYTHON_VERSION="3.9.6"
> Step 2/15 : FROM python:${PYTHON_VERSION}-alpine3.15
> manifest for python:3.9.6-alpine3.15 not found: manifest unknown: manifest unknown
> ```
> 
> We need to use a valid tag from https://hub.docker.com/_/python.
> 
> I think we can use `3.10.2-alpine3.15` which will solve [#19857 (comment)](https://github.com/Azure/azure-cli/issues/19857#issuecomment-1016449222) by the way.

You're right, my bad.

**Review by jiasli — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**